pkgs/top-level/all-packages.nix

```diff
@@ -16568,9 +16568,9 @@ in
   iputils = hiPrio (callPackage ../os-specific/linux/iputils { });
   # hiPrio for collisions with inetutils (ping and tftpd.8.gz)
 
-  iptables = iptables-legacy;
-  iptables-legacy = callPackage ../os-specific/linux/iptables { };
-  iptables-nftables-compat = callPackage ../os-specific/linux/iptables { nftablesCompat = true; };
+  iptables = iptables-nftables-compat;
+  iptables-legacy = callPackage ../os-specific/linux/iptables { nftablesCompat = false; };
```

Comment:
From my understanding `iptables-legacy` is incompatible when used together with `iptables` as they both use different backends in the kernel. should we expose `iptables-legacy` in this case at all. If somebody is using `ferm` and also have docker containers with iptables this would lead to breakages no?

Reply:
`ferm` and `shorewall` doesn't support `iptables-nftables` and [never](https://github.com/MaxKellermann/ferm/issues/35#issuecomment-386091563) [will](https://sourceforge.net/p/shorewall/mailman/message/35492188/) so yes, it will break.

Reply:
They do not support the nftables syntax. Do they support the `iptables` wrapper binary though?

Reply:
I don't see why not, but they also use `ipset` which uses old kernel backend. Mixing those two would surely break things.

Reply:
There is other software that uses `ipset` and is also not compatible with `iptables-nft`.
https://github.com/firehol/firehol/issues/352
https://github.com/rancher/k3s/issues/1114
May be a good idea to wait for at least `k3s` to be fixed before merging this.

Reply:
As far as I can see in both `ferm` and `shorewall` `ipset` is optional. We should remove `ipset` from both in that case.

Reply:
For ferm, wouldn't this mean `ferm` firewalls currently using `ipset` (which is possible to specify in ferm syntax, see http://ferm.foo-projects.org/download/2.0/ferm.html (`set`) will break?

Didn't check the other tools.

Reply:
I've updated `ferm` and `shorewall`. `ferm` test is passing but `shorewall` doesn't have a module and test so not sure if it's working. Added some release notes too.
Still not sure what to do with `k3s` but it seems [`ipset` functionality could be disabled](https://github.com/rancher/k3s/issues/959#issuecomment-571903142) too.